File: 2018-03-07/Adapters/app/src/main/java/br/ufpe/cin/if1001/adapters/CustomAdapterActivity.java
package br.ufpe.cin.if1001.adapters;

import android.app.ListActivity;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.TextView;

public class CustomAdapterActivity extends ListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        PessoaAdapter pessoaAdapter = new PessoaAdapter(this, Constants.maisPessoasAinda);
        setListAdapter(pessoaAdapter);
    }

    class PessoaAdapter extends BaseAdapter {

        Context c;
        Pessoa[] pessoas;

        public PessoaAdapter(Context c, Pessoa[] ps) {
            this.c = c;
            this.pessoas = ps;
        }

        @Override
        public int getCount() {
            return pessoas.length;
        }

        @Override
        public Object getItem(int position) {
            return pessoas[position];
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            View v = LayoutInflater.
                    from(c).
                    inflate(R.layout.customadapter, parent, false);

            //Alternativa seria:
            /*
            //View v;
            if( convertView == null) {
                v = LayoutInflater.from(c).inflate(R.layout.customadapter, parent, false);
            } else {
                v = convertView;
            }
            //ainda precisaria dar findViewById... (custoso)
            /**/


            //Buscando a referência ao TextView para inserirmos o nome
            TextView tvNome = v.findViewById(R.id.nome);
            //Buscando a referência ao TextView para inserirmos o login
            TextView tvLogin = v.findViewById(R.id.login);

            //getItem retorna Object, então precisamos dar o cast
            // ou criar um método específico que retorne pessoa
            String nome = ((Pessoa) getItem(position)).getNome();
            String login = ((Pessoa) getItem(position)).getLogin();

            //Efetivamente setando o nome e login na View
            tvNome.setText(nome);
            tvLogin.setText("(" + login +")");

            return v;
        }
    }

}

File: 2019-03-13/Adapters/app/src/main/java/br/ufpe/cin/if710/adapters/CustomAdapterActivity.kt
package br.ufpe.cin.if710.adapters

import android.app.ListActivity
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.BaseAdapter
import android.widget.TextView

//Herda de ListActivity
class CustomAdapterActivity : ListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val pessoaAdapter = PessoaAdapter(this, Constants.maisPessoasAinda)

        //chamada ao setListAdapter da classe ListActivity
        listAdapter = pessoaAdapter

    }

    internal inner class PessoaAdapter(
            var c: Context,
            var pessoas: Array<Pessoa>) : BaseAdapter() {

        override fun getCount(): Int {
            return pessoas.size
        }

        override fun getItem(position: Int): Any {
            return pessoas[position]
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            //val v = LayoutInflater.from(c).inflate(R.layout.item_custom_adapter, parent, false)

            //Alternativa seria:
            val v : View
            //View v;
            if( convertView == null) {
                v = LayoutInflater.from(c).inflate(R.layout.item_custom_adapter, parent, false);
            } else {
                v = convertView;
            }
            //ainda precisaria dar findViewById... (custoso)
            /**/


            //Buscando a referência ao TextView para inserirmos o nome
            val tvNome = v.findViewById<TextView>(R.id.nome)
            //Buscando a referência ao TextView para inserirmos o login
            val tvLogin : TextView = v.findViewById(R.id.login)

            //getItem retorna Object, então precisamos dar o cast
            // ou criar um método específico que retorne pessoa
            val pessoa = getItem(position) as Pessoa
            val nome = pessoa.nome
            val login = pessoa.login

            //Efetivamente setando o nome e login na View
            tvNome.text = nome
            tvLogin.text = "($login)"

            return v
        }
    }
}